src/czar/WorkerIngestProcessor.cc

```diff
@@ -0,0 +1,50 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "czar/WorkerIngestProcessor.h"
+
+using namespace std;
+
+namespace lsst::qserv::czar::ingest {
+
+Request::Request(function<Result()> const& req, shared_ptr<ResultQueue> resultQueue)
+        : _req(req), _resultQueue(resultQueue) {}
+
+void Request::operator()() { _resultQueue->push(_req()); }
```

Comment:
I have a minor complaint about operator()(). It can be very difficult to find all instances where it is called when the ide isn't doing a good job. grep "()" generates vast numbers of false positives.

Reply:
Good point.